# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLChild.h
# Commit: 71c122ac0ca7
# Full Hash: 71c122ac0ca73391866b1ef19f4f82bc2d28568b
# Author: David Parks <davidp99@gmail.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Initial out-of-process WebGL implementation. r=mccr8,handyman
#   
#   Splits WebGLContext into ClientWebGLContext and HostWebGLContext.  The Client enables the JS-control of a WebGL context in a content procecss while the Host executes the WebGL graphics operations (via a WebGLContext that maintains much of the existing code) in the compositor process.  At this point, the cross-process behavior is disabled -- this series of patches is an incremental step toward that final goal.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54018
# ==============================================================================

diff -r 5cbf5bb11c58 -r 71c122ac0ca7 dom/canvas/WebGLChild.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/canvas/WebGLChild.h	Wed Jan 08 22:19:14 2020 +0000
@@ -0,0 +1,33 @@
+/* -*- Mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef WEBGLCHILD_H_
+#define WEBGLCHILD_H_
+
+#include "mozilla/dom/PWebGLChild.h"
+#include "nsWeakReference.h"
+
+namespace mozilla {
+
+class ClientWebGLContext;
+
+namespace dom {
+
+class WebGLChild : public PWebGLChild {
+ public:
+  mozilla::ipc::IPCResult RecvQueueFailed();
+  mozilla::ClientWebGLContext* GetContext();
+
+ protected:
+  friend mozilla::ClientWebGLContext;
+  void SetContext(mozilla::ClientWebGLContext* aContext);
+
+  nsWeakPtr mContext;
+};
+
+}  // namespace dom
+}  // namespace mozilla
+
+#endif  // WEBGLCHILD_H_